### PR TITLE
Modify stack size flag for windows

### PR DESCRIPTION
**Pull request by TheGreatRambler**
`add_link_flag("-Wl,--stack,8388608")` is used whenever the build script detects a non MSVC compiler, but the clang compiler is not considered a MSVC compiler (at least in the MSYS shell) and this flag causes an error in clang. Checking for clang before applying this flag fixes the issue

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -270,7 +270,11 @@ else()
   if(WIN32)
     add_compile_flag("-D_GNU_SOURCE")
     add_compile_flag("-D__STDC_FORMAT_MACROS")
-    add_link_flag("-Wl,--stack,8388608")
+    if(NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+      add_link_flag("-Wl,--stack,8388608")
+    else()
+      add_link_flag("-Wl,-stack_size,8388608")
```
[the remark this answers is not in the document]
Comment (TheGreatRambler) [reply CMakeLists.txt:276]: I believed there was a good reason to keep the flag around, though 8mbs is the default on windows so odd to explicitly define it

[the remark this answers is not in the document]
Comment (TheGreatRambler) [reply CMakeLists.txt:276]:
```
-- The C compiler identification is Clang 14.0.0 with GNU-like command-line
-- The CXX compiler identification is Clang 14.0.0 with GNU-like command-line
lld-link: warning: ignoring unknown argument '--stack'
```

[the remark this answers is not in the document]
Comment (TheGreatRambler) [reply CMakeLists.txt:276]:
```
lld-link: warning: ignoring unknown argument '-stack_size'
```
Well the second commit was a mistake

[the remark this answers is not in the document]
Comment (sbc100) [reply CMakeLists.txt]: > I believed there was a good reason to keep the flag around, though 8mbs is the default on windows so odd to explicitly define it

I think 1Mb is the default on windows: https://docs.microsoft.com/en-us/cpp/build/reference/stack-stack-allocations?view=msvc-170

[the remark this answers is not in the document]
Comment (sbc100) [reply CMakeLists.txt]: > ```
> -- The C compiler identification is Clang 14.0.0 with GNU-like command-line
> -- The CXX compiler identification is Clang 14.0.0 with GNU-like command-line
> lld-link: warning: ignoring unknown argument '--stack'
> ```

Looks like lld-link accepts `/stack:xxx`   .. and I think MSVC too.. so perhaps that answer it do change that line from `-Wl,--stack,8388608` to `-Wl,-/stack:8388608`?  Then i could work for both MSVC and clang?

[the remark this answers is not in the document]
Comment (TheGreatRambler) [reply CMakeLists.txt]: That did actually work, but wouldn't this be triggered on Clang for Linux?

[the remark this answers is not in the document]
Comment (sbc100) [reply CMakeLists.txt]: This block is WIN32 only.

[the remark this answers is not in the document]
Comment (TheGreatRambler) [reply CMakeLists.txt]: Oh you're absolutely right

File: CMakeLists.txt [hunk 2]
```diff
@@ -270,7 +270,11 @@ else()
   if(WIN32)
     add_compile_flag("-D_GNU_SOURCE")
     add_compile_flag("-D__STDC_FORMAT_MACROS")
-    add_link_flag("-Wl,--stack,8388608")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+      add_link_flag("-Wl,/stack:8388608")
```
Comment (sbc100) [inline CMakeLists.txt:274]: I think this will work in both cases won't it?  No need for an if here.

Comment (TheGreatRambler) [reply CMakeLists.txt:274]: This was originally a MinGW path, right? `/stack` would be interpreted as a file path on GCC.

Comment (sbc100) [reply CMakeLists.txt:274]: I'm not sure.. I'm pretty sure `--stack` is not a valid GNU ld linker flag.   What linker to you think it was meant for?

Comment (sbc100) [reply CMakeLists.txt:274]: Hmm.. seems you are right.. it looks like the MinGW version of ld does tack `--stack`

**Review by sbc100 — APPROVED** [review]

**Pull request merged**